src/main/scala/io/scalac/amqp/impl/ExchangeSubscriber.scala

```diff
@@ -7,36 +7,91 @@ import com.rabbitmq.client.Channel
 import io.scalac.amqp.Routed
 import org.reactivestreams.{Subscriber, Subscription}
 
+import scala.annotation.tailrec
+import scala.collection.immutable.Queue
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
+import scala.concurrent.stm.{Ref, atomic}
+import scala.util.control.NonFatal
 
 private[amqp] class ExchangeSubscriber(channel: Channel, exchange: String)
   extends Subscriber[Routed] {
   require(exchange.length <= 255, "exchange.length > 255")
 
   val active = new AtomicReference[Subscription]()
+  val publishingThreadRunning = Ref(false)
+  val buffer = Ref(Queue[Routed]())
+  var channelOpen = true
```

Comment:
It seems that access to this variable is not synchronized anywhere. I'm right?
IMO This is more of a `closeRequested` than indicator of whether channel is closed or not.